### ISSUE TITLE
Broken interaction with RStudio -> Build -> Issues pane and new testthat errors

### ISSUE THREAD

**Issue by DavisVaughan** [rstudio/rstudio#8505]
See https://github.com/r-lib/testthat/issues/1220 for original report

Something seems to have happened with the RStudio->Build->Issues pane when running testthat. I run tests with CMD+SHIFT+T and normally I can click the Issues pane and quick jump to test failures. But when I try to double click here I get a “No such file or directory” error. The line numbers look a little garbled, could something have broken in testthat?

This error happens with testthat 3.0.0, but testthat 2.3.2 works fine when I revert back to it.

![Screen Shot 2020-11-03 at 11 24 19 AM](https://user-images.githubusercontent.com/19150088/98013325-3ed8ea80-1dc8-11eb-9f11-a16da7d73f49.png)

**Comment by kevinushey** [comment]
@jmcphers: what's your thought on solving this in 1.4 versus 1.4-Juliet-Rose? The underlying problem is that our parsing of paths from testthat test output is broken right now, due to the way testthat now formats its output.

I' leaning towards fixing it for Juliet Rose, but have parked in 1.4 for now.

@DavisVaughan @hadley: is there any way RStudio could request the "old" style testthat output so we could have a safer fix for 1.4?

**Comment by hadley** [comment]
RStudio could request both the default and a standard reporter, and then parse the info from the standard reporter? (There is the existing `testthat::RstudioReporter`; I don't remember why we created it or if it's used anywhere, but it seems like the right hook).

**Comment by jmcphers** [comment]
Much as it pains me to say it, this is pretty advanced usage and easily worked around by using the older version of testthat (plus the error line number is right there so you can go there manually; the double clicking is just a convenience). I think this belongs in Juliet Rose, but we could consider developing a fix soon and applying it to Wax Begonia if it winds up being extremely small and safe.

**Comment by hadley** [comment]
(The other route to a fix would be via #1941 — if that was implemented then the linking could be entirely testthat's responsibility)

**Comment by kevinushey** [comment]
FWIW here's where we build the command to be used to run tests:

https://github.com/rstudio/rstudio/blob/ebb8ff189b541c3b704d7b49df5cd0f5738796d5/src/cpp/session/modules/build/SessionBuild.cpp#L1104-L1106

We could consider (silently) adding something like `reporter = "RstudioReporter"`, if that reporter would behave better.

**Comment by hadley** [comment]
You could do:

```R
devtools::test(
  reporter = MultiReporter$new(list(
    testthat::find_reporter(testthat::default_reporter()), 
    RstudioReporter$new(file = "test-results.txt")
  ))
)
```

That will run the default reporter and use the RStudio reporter to save the results in standard format to a known location. You'd need to double check that it gives you the details you want.

I'm reasonably sure that code hasn't changed in a long time, so should work with most versions of testthat found in the wild.

**Comment by mikebessuille** [comment]
need to make sure we support old and new versions

**Comment by kevinushey** [comment]
Will be fixed in 1.4-JR via https://github.com/rstudio/rstudio/issues/8505.

**Comment by astayleraz** [comment]
Verified in Version 1.4.1514-2 and R 3.6.0 that this looks to be working as expected.
